File: problem_solving_with_DSA/array/C/removeDuplicate.c
#include <stdio.h>

/**
 * rmDuplicates - Remove duplicate from a sorted array
 * 
 * @arr: Sorted array
 * @n: Size of the array 
 * Return: Size of the new array 
 */
int rmDuplicates(int arr[], int n)
{
    int len = 0;
    
    for (int i = 0; i < n-1; i++)
    {
        if (arr[i] != arr[i+1])
        {
            arr[len++] = arr[i];
        }
    }
    arr[len++] = arr[n-1];
    
    return (len);
}

/**
 * printArray - Print array 
 * 
 * @arr: Array to print
 * @n: Size of the array
 */
void printArray(int arr[], int n)
{
    printf("{");
    for (int i = 0; i < n; i++)
    {
        if (i < n-1)
            printf("%d, ", arr[i]);
        else
            printf("%d}\n", arr[i]);
    }
}



int main(void)
{
    int arr[] = {1,1,1,3,3,5,5};
    int n = 7;
    
    printArray(arr, n);
    int newN = rmDuplicates(arr, n);
    printf("%d\n", newN);
    printArray(arr, newN);
    
    return (0);
}

File: problem_solving_with_DSA/array/python/removeDuplicate.py
"""
An in-place algorithm to Remove duplicate in a sorted array
"""

def rmDuplicates(arr ,n):
    """Remove duplicate in a sorted array
    
    Args:
        arr (arr[int]): Sorted array
        n (int) -  Size of the array
        
    Return: length of the new array
    """
    if n==0 or n==1:
        return n
    
    len = 0
    for i in range(n-1):
        if arr[i] != arr[i+1]:
            arr[len] = arr[i]
            len += 1
    arr[len] = arr[n-1]
    len += 1
    return len
    
def printArray(arr, n):
    """Print array
    
    Args:
        arr (arr[int]): array
        n (int) -  Size of the array
    """
    print('[', end="")
    for i in range(n):
        if i < n-1:
            print("{}, ".format(arr[i]), end="")
        else:
            print("{}]".format(arr[i]))

if __name__ == '__main__':
    input = [1, 1, 1, 2, 2, 2, 3, 3]
    len = 8
    
    printArray(input, len)
    newLen = rmDuplicates(input, len);
    printArray(input, newLen)
